tanjun/_vendor/__init__.py

```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
+
+__all__ = ["inspect"]
+
+import sys
+
+
+if sys.version_info >= (3, 10):
+    import inspect
+
+else:
+    from . import inspect
```

Comment:
Maybe move the UNION_TYPES set here so it can be used from multiple places at once?